ctapipe/visualization/bokeh.py

```diff
@@ -1,408 +1,634 @@
-            self.cm.high = val.item()
+    x_offset = width[:, np.newaxis] * np.array([-1, -1, 1, 1])
+    y_offset = width[:, np.newaxis] * np.array([1, -1, -1, 1])
 
-    @property
-    def active_pixels(self):
-        return self._active_pixels
-
-    @active_pixels.setter
-    def active_pixels(self, listval):
-        self._active_pixels = listval
-
-        palette = palettes.Set1[9]
-        palette = tuple([palette[0]] + list(palette[3:]))
-        self.active_colors = [palette[i % (len(palette))] for i in range(len(listval))]
-        self.highlight_pixels()
-
-    def reset_pixels(self):
-        self.active_pixels = [0] * len(self.active_pixels)
-
-    def _draw_camera(self):
-        # TODO: Support other pixel shapes OR switch to ellipse
-        # after https://github.com/bokeh/bokeh/issues/6985
-        self.glyphs = self.fig.ellipse(
-            "x",
-            "y",
-            color="image",
-            width="width",
-            height="height",
-            line_color="outline_color",
-            line_alpha="outline_alpha",
-            line_width=2,
-            nonselection_fill_color="image",
-            nonselection_fill_alpha=1,
-            nonselection_line_color="outline_color",
-            nonselection_line_alpha="outline_alpha",
-            source=self.cdsource,
+    xs = x[:, np.newaxis] + x_offset
+    ys = y[:, np.newaxis] + y_offset
+    return xs, ys
+
+
+class BokehPlot:
+    """Base class for bokeh plots"""
+
+    def __init__(self, use_notebook=None, **figure_kwargs):
+        # only use autoshow / use_notebook by default if we are in a notebook
+        self._use_notebook = use_notebook if use_notebook is not None else is_notebook()
+        self._handle = None
+        self.figure = figure(**figure_kwargs)
+
+        if figure_kwargs.get("match_aspect"):
+            # Make sure the box zoom tool does not distort the camera display
+            for tool in self.figure.toolbar.tools:
+                if isinstance(tool, BoxZoomTool):
+                    tool.match_aspect = True
+
+    def show(self):
+        """Display the figure"""
+        if self._use_notebook:
+            output_notebook()
+        else:
+            # this only sets the default name, created only when show is called
+            output_file(tempfile.mktemp(prefix="ctapipe_bokeh_", suffix=".html"))
+
+        self._handle = show(self.figure, notebook_handle=self._use_notebook)
+
+    def update(self):
+        """Update the figure"""
+        if self._use_notebook and self._handle:
+            push_notebook(handle=self._handle)
+
+
+class CameraDisplay(BokehPlot):
+    """
+    CameraDisplay implementation in Bokeh
+    """
+
+    def __init__(
+        # same options as MPL display
+        self,
+        geometry: CameraGeometry = None,
+        image=None,
+        cmap="inferno",
+        norm="lin",
+        autoscale=True,
+        title=None,
+        # bokeh specific options
+        use_notebook=None,
+    ):
```

Comment:
missing docstring here. In particular, use_notebook needs to be defined as it is not in the MPL version and None has a special meaning (to auto-detect)